### PR TITLE
Update settings on template activation

### PR DESCRIPTION
**Pull request by doekenorg**
This PR addresses #2117 

After activating or installing a plugin, the selectors are updated; and the settings block is re-rendered and replaced. This will ensure all the new settings are available.

💾 [Build file](https://www.dropbox.com/scl/fi/r845y7ujndgz7pjuztop9/gravityview-2.29-12afca1a4.zip?rlkey=iy270yx0yabawfamj6za0esvr&dl=1) (12afca1a4).

### CODE REVIEW

**Comment by mrcasual** [comment]
@doekenorg, is this PR ready for review/testing? If so, please update the Project information, `@` tag folks, etc.

**Comment by doekenorg** [comment]
I updated the connected issue, but I have now updated this one as well. I'm not sure which I should update now. Doing both sounds too much.

File: assets/js/admin-views.js [hunk 1]
```diff
@@ -2950,7 +2998,7 @@
 
 			viewGeneralSettings.metaboxObj
 				// What happens after tabs are generated
-				.on( 'tabscreate', viewGeneralSettings.tabsCreate )
+				// .on( 'tabscreate', viewGeneralSettings.tabsCreate )
```
Comment (zackkatz) [inline assets/js/admin-views.js:3001]: Remove unused code

File: assets/js/admin-views.js [hunk 2]
```diff
@@ -2897,6 +2931,20 @@
...
+		/**
+		 * Refreshes the tabs after HTML was replaced.
+		 *
+		 * @since $ver$
+		 */
+		refresh: function () {
+			viewGeneralSettings.metaboxObj.trigger( 'change' );
+
+			viewGeneralSettings.metaboxObj
+				// .off( 'tabscreate', viewGeneralSettings.tabsCreate )
```
Comment (zackkatz) [inline assets/js/admin-views.js:2943]: Remove unused code

**Review by zackkatz — CHANGES_REQUESTED** [review]
Looks good code-wise @doekenorg 👍 Just a couple commented-out lines.

File: assets/js/admin-views.js (hunk 2 above)
Comment (doekenorg) [reply assets/js/admin-views.js:2943]: Wasn't sure if I could remove this, but it looks like I can.

**Comment by mrcasual** [comment]
@Mwalek, please test.

**Comment by Mwalek** [comment]
@doekenorg @mrcasual looks good! I tested with PHP versions 7.4 & 8.3.

While testing this I found a few minor issues that are indirectly related to this functionality. @mrcasual before I create GH issues for them, could you please have a look and see if they're all things we want to fix?

Loom Video: https://www.loom.com/share/0665547e6d8b4a7dbb500ef8fb0bb618

- In the default template selection layout, aside from the outline on the button, there is no indication that the plugin is installing.
- Clicking "Install and Activate" or "Activate" from the placeholder behaves differently than performing the same action from the dropdown. The page is reloaded and data is lost.
- Placeholders in the settings menu show the "Upgrade" option when hovered, even though the All Access license is active. It should display "Install/Activate" instead.
- When using the placeholder to activate, the user is prompted twice for confirmation about leaving the page: once via the page itself and once through a browser modal.

**Comment by mrcasual** [comment]
@Mwalek, the status indicator, upgrade badge, and confirmation messages are minor issues that should be corrected. What's more concerning is that when you click to install and use DataTables, the View type dropdown still shows Table.

In the order of priority:

1) When creating the View and selecting the View type, the View type of the Multiple Entries layout should be the same.
2) Loading indicator should be displayed when installing the layout in the View type selection during View creation
3) Upgrade badge + confirmation message (these can also be resolved under a separate PR)

@doekenorg, over to you.

**Comment by doekenorg** [comment]
I've created https://github.com/GravityKit/GravityView/issues/2158 to address the other issues.

**Comment by doekenorg** [comment]
@Mwalek Back to you for (hopefully final) testing.

**Comment by Mwalek** [comment]
@doekenorg thank you! I retested this and can confirm that the following issues are resolved by the latest build:

>1) When creating the View and selecting the View type, the View type of the Multiple Entries layout should be the same.
>2) Loading indicator should be displayed when installing the layout in the View type selection during View creation

**Pull request merged**